projecttacoma/deqm-test-server
# X prov map

# Summary
Added functionality for mapping X-Provenance headers to an AuditEvent resource and storing that resource in our mongo db for $submit-data requests

## New behavior
When sending a submit-data request, if you add the optional X-Provenance header, an AuditEvent will be created with the appropriate properties, documenting the action that occurred and the agent that was responsible

## Code changes
- Added new provenanceUtils.js file with function for mapping headers to AuditEvent Object
- Editted the existing submitData function to create the AuditEvent resource when necessary
- Added testing and fixtures for new functions

# Testing guidance
- Run all tests
- Connect to robomongo
- Try sending a `$submit-data` request (more on that [here](https://github.com/projecttacoma/deqm-test-server/pull/7)) with X-Provenance headers. I'll post a couple of options in Slack which have worked for me
- Ensure that AuditEvent resources are created correctly displaying the information stored in the X-Provenance Header
- Here's the [AuditEvent IG](http://www.hl7.org/fhir/auditevent.html). Please look through the documentation and ensure that all fields are of the correct cardinality and are mapped correctly.
